### PR TITLE
Fix build errors

### PR DESCRIPTION
**Pull request by bovine3dom**
Typescript complains about us giving arguments to super(). It seems to build fine if I remove the arguments and the autocmds still work.

Anyone have any ideas if it's real?

### CODE REVIEW

File: src/lib/requests.ts [hunk 1]
```diff
@@ -9,7 +9,8 @@ const logger = new Logger("requests")
 
 class DefaultMap extends Map {
     constructor(private defaultFactory, ...args) {
-        super(...args)
+        //super(...args)
+        super()
```
Comment (bovine3dom) [inline src/lib/requests.ts]: This is the line in question.

Comment (glacambre) [reply src/lib/requests.ts]: Just realized that Map() takes a [single argument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), so
```suggestion
constructor(private defaultFactory, args) {
        super(args)
```
Should preserve original behavior and be more correct.

Comment (bovine3dom) [reply src/lib/requests.ts]: Cool. Can't use that suggestion as it only deletes one line. Will do it the old fashioned way.

Comment (glacambre) [reply src/lib/requests.ts]: I have no idea how to remove multiple lines with a single suggestion, do you know if it's possible?

Comment (bovine3dom) [reply src/lib/requests.ts]: It isn't possible. Make multiple suggestions. As I said, I'll do it the old fashioned way, or you can, I think you can push to this branch.

Comment (bovine3dom) [reply src/lib/requests.ts]:
```
Successfully installed Tridactyl native messenger!
Run ':native' in Firefox to check.
ℹ ｢atl｣: Using typescript@3.3.3333 from typescript
ℹ ｢atl｣: Using tsconfig.json from /home/olie/projects/tridactyl/tsconfig.json
ℹ ｢atl｣: Checking started in a separate process...
✖ ｢atl｣: Checking finished with 1 errors

ERROR in [at-loader] ./src/lib/requests.ts:12:9
    TS2554: Expected 0 arguments, but got 1.

```

No cigar :(

Comment (glacambre) [reply src/lib/requests.ts]: That's weird. I guess the solution is to go back to prototype-based definitions and do something like this:
```
let DefaultMap = function (defaultFactory, ...args) {
  Map.prototype.constructor.call(this, args)
  this.defaultFactory = defaultFactory;
}
DefaultMap.prototype = Object.create(Map.prototype)
DefaultMap.prototype.constructor = DefaultMap
DefaultMap.prototype.get = function (key) {
  if (!this.has(key)) {
    this.set(key, this.defaultFactory(key))
  }
  return Map.prototype.get.call(this, key)
}
```

**Pull request merged**